localization/yabloc/yabloc_particle_filter/src/camera_corrector/camera_particle_corrector_core.cpp

```diff
@@ -0,0 +1,324 @@
+  }
+
+  return {reliable_cloud, good_cloud};
+}
+
+void CameraParticleCorrector::on_line_segments(const PointCloud2 & line_segments_msg)
+{
+  tier4_autoware_utils::StopWatch stop_watch;
+  const rclcpp::Time stamp = line_segments_msg.header.stamp;
+  std::optional<ParticleArray> opt_array = this->get_synchronized_particle_array(stamp);
+  if (!opt_array.has_value()) {
+    return;
+  }
+
+  const rclcpp::Duration dt = (stamp - opt_array->header.stamp);
+  if (std::abs(dt.seconds()) > 0.1) {
+    const std::string text = "Timestamp gap between image and particles is LARGE ";
+    RCLCPP_WARN_STREAM(get_logger(), text << dt.seconds());
+  }
+
+  auto [line_segments_cloud, iffy_line_segments_cloud] = split_line_segments(line_segments_msg);
+  ParticleArray weighted_particles = opt_array.value();
+
+  bool publish_weighted_particles = true;
+  const Pose meaned_pose = mean_pose(weighted_particles);
+  {
+    // Check travel distance and publish weights if it is enough long
+    Eigen::Vector3f mean_position = common::pose_to_affine(meaned_pose).translation();
+    if ((mean_position - last_mean_position_).squaredNorm() > 1) {
+      last_mean_position_ = mean_position;
+    } else {
+      using namespace std::literals::chrono_literals;
+      publish_weighted_particles = false;
+      RCLCPP_WARN_STREAM_THROTTLE(
+        get_logger(), *get_clock(), 2000,
+        "Skip particle weighting due to almost the same position");
+    }
+  }
+
+  cost_map_.set_height(meaned_pose.position.z);
+
+  if (publish_weighted_particles) {
+    for (auto & particle : weighted_particles.particles) {
+      Sophus::SE3f transform = common::pose_to_se3(particle.pose);
+      LineSegments transformed_line_segments =
+        common::transform_line_segments(line_segments_cloud, transform);
+      LineSegments transformed_iffy_line_segments =
+        common::transform_line_segments(iffy_line_segments_cloud, transform);
+      transformed_line_segments += transformed_iffy_line_segments;
+
+      float logit = compute_logit(transformed_line_segments, transform.translation());
+      particle.weight = logit_to_prob(logit, 0.01f);
+    }
+
+    if (enable_switch_) {
+      this->set_weighted_particle_array(weighted_particles);
+    }
+  }
+
+  cost_map_.erase_obsolete();  // NOTE:
+  pub_marker_->publish(cost_map_.show_map_range());
+
+  // DEBUG: just visualization
+  {
+    Pose meaned_pose = mean_pose(weighted_particles);
+    Sophus::SE3f transform = common::pose_to_se3(meaned_pose);
+
+    pcl::PointCloud<pcl::PointXYZI> cloud = evaluate_cloud(
+      common::transform_line_segments(line_segments_cloud, transform), transform.translation());
+    pcl::PointCloud<pcl::PointXYZI> iffy_cloud = evaluate_cloud(
+      common::transform_line_segments(iffy_line_segments_cloud, transform),
+      transform.translation());
+
+    pcl::PointCloud<pcl::PointXYZRGB> rgb_cloud;
+    pcl::PointCloud<pcl::PointXYZRGB> rgb_cloud2;
```

Comment:
How about `rgb_cloud2` -> `iffy_rgb_cloud`?

Reply:
The suggestion look good. Your review is reflected in d9617f3. :+1: